Source/BCJSONDecoder.m

```
@@ -46,25 +47,41 @@ + (id)decodeFileAtURL:(NSURL*)url error:(NSError* __autoreleasing*)error {
 - (instancetype)initWithRootObject:(NSDictionary*)root {
   self = [super init];
   if (self) {
-    if (!root || ![root isKindOfClass:[NSDictionary class]] ) {
-      NSAssert( NO, @"Not a dictionary" );
-      [self setErrorWithDescription:@"Invalid JSON archive"];
-    } else {
-      self.root = root;
-    }
+    self.root = root;
   }
   
   return self;
 }
 
 - (instancetype)initWithData:(NSData *)data {
+  self = [super init];
+  if (self) {
+    self.data = data;
+  }
+  return self;
+}
+
+- (void)setRoot:(NSDictionary*)root {
+  if (!root || [root isKindOfClass:[NSDictionary class]]) {
```

If something passes in a non-dictionary object here, would it be better to throw an exception? Generally I'd say that's better than quietly swallowing the mistake.